lib/ansible/modules/notification/cisco_spark.py

```diff
@@ -0,0 +1,205 @@
+    require: true
+    aliases: ['token']
+
+  message:
+    description:
+      - The message you would like to send.
+    required: True
+'''
+
+EXAMPLES = """
+# Note: The following examples assume a variable file has been imported
+# that contains the appropriate information.
+
+- name: Cisco Spark - Markdown Message to a Room
+  cisco_spark:
+    recipient_type: roomId
+    recipient_id: "{{ room_id }}"
+    message_type: markdown
+    personal_token: "{{ token }}"
+    message: "**Cisco Spark Ansible Module - Room Message in Markdown**"
+
+- name: Cisco Spark - Text Message to a Room
+  cisco_spark:
+    recipient_type: roomId
+    recipient_id: "{{ room_id }}"
+    message_type: text
+    personal_token: "{{ token }}"
+    message: "Cisco Spark Ansible Module - Room Message in Text"
+
+- name: Cisco Spark - Text Message by an Individuals ID
+  cisco_spark:
+    recipient_type: toPersonId
+    recipient_id: "{{ person_id}}"
+    message_type: text
+    personal_token: "{{ token }}"
+    message: "Cisco Spark Ansible Module - Text Message to Individual by ID"
+
+- name: Cisco Spark - Text Message by an Individuals E-Mail Address
+  cisco_spark:
+    recipient_type: toPersonEmail
+    recipient_id: "{{ person_email }}"
+    message_type: text
+    personal_token: "{{ token }}"
+    message: "Cisco Spark Ansible Module - Text Message to Individual by E-Mail"
+
+"""
+
+RETURN = """
+status_code:
+  description:
+    - The Response Code returned by the Spark API
+    - Full Responsde Code explanations can be found at https://developer.ciscospark.com/endpoint-messages-post.html
+  returned: always
+  type: int
+  sample: 200
+
+message:
+    description:
+      - The Response Message returned by the Spark API
+      - Full Responsde Code explanations can be found at https://developer.ciscospark.com/endpoint-messages-post.html
+    returned: always
+    type: string
+    sample: OK (585 bytes)
+"""
+
+
+def spark_message(module):
+    """ When check mode is specified, establish a read only connection, that does not return any user specific
+    data, to validate connectivity. In regular mode, send a message to a Cisco Spark Room or Individual"""
+
+    # Ansible Specific Variables
+    results = {}
+    ansible = module.params
+
+    headers = {
+        'Authorization': 'Bearer {}'.format(ansible['personal_token']),
+        'content-type': 'application/json'
+    }
+
+    if module.check_mode:
+
+        url = "https://api.ciscospark.com/v1/people/me"
+
+        response, info = fetch_url(module, url, headers=headers)
+
+    else:
+
+        url = "https://api.ciscospark.com/v1/messages"
+
+        # By Room ID
+        payload = {
+            ansible['recipient_type']: ansible['recipient_id'],
+            ansible['message_type']: ansible['message']
+        }
+
+        payload = module.jsonify(payload)
+
+        response, info = fetch_url(module, url, data=payload, headers=headers)
```

Comment:
The default payload (when not provided) is None, so if you would truly prefer to be explicit, you'd be using my example instead of rewriting and duplicating the fetch_url statement. (Which I specifically wanted to avoid...)

```python
    if module.check_mode:
        url = "https://api.ciscospark.com/v1/people/me"
        payload = None
    else:
        url = "https://api.ciscospark.com/v1/messages"
        payload = {
            ansible['recipient_type']: ansible['recipient_id'],
            ansible['message_type']: ansible['message']
        }
        payload = module.jsonify(payload)

    response, info = fetch_url(module, url, data=payload, headers=headers)
```